server.js

```diff
@@ -1,9 +1,26 @@
+require("dotenv").config()
+require("colors")
 const express = require("express")
+const session = require("express-session")
 const app = express()
+const connectDB = require("./db/connect")
+const PORT = process.env.PORT || 8080
+const taskRouter = require("./routes/tasks")
+const setMessage = require("./middleware/message")
 
 // set the view engine to ejs
 app.set("view engine", "ejs")
 
+app.use(
+  session({
+    secret: process.env.SESSION_SECRET,
```

Comment:
Validate it's present at startup if it's a required environment variable!

I harp on this because in a cloud environment, an app that crashes on startup generally won't be deployed. The process exits, it's easy for any system to know that the server is dead and broken, and that can block a deployment in docker or Kubernetes. If your app starts and the server binds, but then fails on every request, you might end up with code in production that sporadically sends errors for most requests!

This is just one example of a universal programming rule: when your program is in an invalid state, fail early and fail loudly!

Reply:
> If your app starts and the server binds, but then fails on every request, you might end up with code in production that sporadically sends errors for most requests!

oh wow! that makes so much sense!
will fix it!

Reply:
What would be the best way to do this check? set all the env variables into an object and check each of them using an external library like https://www.npmjs.com/package/joi or check each one for just having any set value in it?

Reply:
I thought I replied to this somewhere but now don't see it so sorry if this is a duplicate. 

Just check for the presence of these values at startup before you call `app.listen`

Reply:
> I thought I replied to this somewhere but now don't see it so sorry if this is a duplicate.
> 
> Just check for the presence of these values at startup before you call `app.listen`

you did! on slack! 😬